File: dotnet/src/demos/dataFlowAnalyzer/dataflow/model/xml/column.cs
namespace gudusoft.gsqlparser.demos.dlineage.dataflow.model.xml
{
    using System;
    using System.Xml.Serialization;

    public class column
	{

        [XmlAttribute]
        public string name { get; set; }

        [XmlAttribute]
        public string id { get; set; }

        [XmlAttribute]
        public string coordinate { get; set; }

        public virtual Tuple<int, int> StartPos
        {
            get
            {
                return PositionUtil.getStartPos(coordinate);
            }
        }

        public virtual Tuple<int, int> EndPos
        {
            get
            {
                return PositionUtil.getEndPos(coordinate);
            }
        }


    }

}

File: java/src/demos/dlineage/dataflow/model/xml/column.java

package demos.dlineage.dataflow.model.xml;

import org.simpleframework.xml.Attribute;

import demos.dlineage.util.Pair;

public class column
{

	@Attribute(required = false)
	private String name;

	@Attribute(required = false)
	private String id;

	@Attribute(required = false)
	private String coordinate;

	public String getCoordinate( )
	{
		return coordinate;
	}

	public Pair<Integer, Integer> getStartPos( )
	{
		return PositionUtil.getStartPos( coordinate );
	}

	public Pair<Integer, Integer> getEndPos( )
	{
		return PositionUtil.getEndPos( coordinate );
	}

	public void setCoordinate( String coordinate )
	{
		this.coordinate = coordinate;
	}

	public String getName( )
	{
		return name;
	}

	public void setName( String name )
	{
		this.name = name;
	}

	public String getId( )
	{
		return id;
	}

	public void setId( String id )
	{
		this.id = id;
	}

}
